x/group/server/keeper.go

```diff
@@ -0,0 +1,542 @@
+package server
+
+import (
+	"fmt"
+	"reflect"
+
+	gogotypes "github.com/gogo/protobuf/types"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	paramstypes "github.com/cosmos/cosmos-sdk/x/params/types"
+
+	"github.com/regen-network/regen-ledger/orm"
+	"github.com/regen-network/regen-ledger/x/group/types"
+)
+
+const (
+	// Group Table
+	GroupTablePrefix        byte = 0x0
+	GroupTableSeqPrefix     byte = 0x1
+	GroupByAdminIndexPrefix byte = 0x2
+
+	// Group Member Table
+	GroupMemberTablePrefix         byte = 0x10
+	GroupMemberByGroupIndexPrefix  byte = 0x11
+	GroupMemberByMemberIndexPrefix byte = 0x12
+
+	// Group Account Table
+	GroupAccountTablePrefix        byte = 0x20
+	GroupAccountTableSeqPrefix     byte = 0x21
+	GroupAccountByGroupIndexPrefix byte = 0x22
+	GroupAccountByAdminIndexPrefix byte = 0x23
+
+	// Proposal Table
+	ProposalTablePrefix               byte = 0x30
+	ProposalTableSeqPrefix            byte = 0x31
+	ProposalByGroupAccountIndexPrefix byte = 0x32
+	ProposalByProposerIndexPrefix     byte = 0x33
+
+	// Vote Table
+	VoteTablePrefix           byte = 0x40
+	VoteByProposalIndexPrefix byte = 0x41
+	VoteByVoterIndexPrefix    byte = 0x42
+)
+
+type Keeper struct {
```

Comment:
I don't believe we can at this point, we need it in `RegenApp` in `app.go`.

Reply:
I don't believe we can at this point, we need it in `x/group/module` `AppModule`.

Reply:
OK, so probably later we will need to refactore it to a new design of MsgServices (see `x/data` for example - there is no Keeper).

Reply:
yeah agreed and I would have followed the same design as `x/data` and `x/ecocredit` if I could but we need to provide a `paramstypes.Subspace` to set max comment length and the current way of registering services doesn't support that.

Reply:
If `paramstypes.Subspace` is available in `NewRegenApp`  then you can do it: 
* when creating a module, you can pass more parameters
* then you can store it in a module structure
* finally, the module object calls `server.RegisterServices` and it can pass that parameters to that function.

Reply:
good point, will try that approach, thanks!